sources/settings-extensions/pki/src/lib.rs

```diff
@@ -0,0 +1,109 @@
+/// Settings related to Custom CA Certificates.
+use bottlerocket_settings_sdk::{GenerateResult, SettingsModel};
+use model_derive::model;
+use modeled_types::{Identifier, PemCertificateString};
+use serde::{Deserialize, Deserializer, Serialize, Serializer};
+use std::{collections::HashMap, convert::Infallible};
+
+#[derive(Debug, Default, PartialEq)]
+pub struct PkiSettingsV1 {
+    pub pki: HashMap<Identifier, PemCertificate>,
+}
+
+impl Serialize for PkiSettingsV1 {
+    fn serialize<S>(&self, serializer: S) -> std::result::Result<S::Ok, S::Error>
+    where
+        S: Serializer,
+    {
+        self.pki.serialize(serializer)
+    }
+}
+
+impl<'de> Deserialize<'de> for PkiSettingsV1 {
+    fn deserialize<D>(deserializer: D) -> std::result::Result<Self, D::Error>
+    where
+        D: Deserializer<'de>,
+    {
+        let pki = HashMap::deserialize(deserializer)?;
+        Ok(Self { pki })
+    }
+}
+
+#[model(impl_default = true)]
+struct PemCertificate {
+    data: PemCertificateString,
+    trusted: bool,
+}
+
+type Result<T> = std::result::Result<T, Infallible>;
+
+impl SettingsModel for PkiSettingsV1 {
+    type PartialKind = Self;
+    type ErrorKind = Infallible;
+
+    fn get_version() -> &'static str {
+        "v1"
+    }
+
+    fn set(_current_value: Option<Self>, _target: Self) -> Result<()> {
+        // Set anything that parses as PkiSettingsV1.
+        Ok(())
+    }
+
+    fn generate(
+        _existing_partial: Option<Self::PartialKind>,
+        _dependent_settings: Option<serde_json::Value>,
+    ) -> Result<GenerateResult<Self::PartialKind, Self>> {
+        Ok(GenerateResult::Complete(PkiSettingsV1 {
+            pki: HashMap::new(),
+        }))
+    }
+
+    fn validate(_value: Self, _validated_settings: Option<serde_json::Value>) -> Result<()> {
+        // Validate anything that parses as PkiSettingsV1.
+        Ok(())
+    }
+}
+
+#[cfg(test)]
+mod test {
+    use super::*;
+
+    static VALID_PEM: &str = include_str!("../tests/data/test-pem");
+
+    #[test]
+    fn test_generate_pki_settings() {
+        assert_eq!(
+            PkiSettingsV1::generate(None, None),
+            Ok(GenerateResult::Complete(PkiSettingsV1 {
+                pki: HashMap::new(),
+            }))
+        )
+    }
+
+    #[test]
+    fn test_serde_pki() {
+        let test_json = format!(
+            r#"{{
```

Comment:
why the double `{` here?

Reply:
Double braces in `format!` are used to produce a JSON string that includes curly braces, and without escaping, this will give errors.